From 1cf023d538f8011ddfbc0320e80133ce4e19f16e Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Sat, 7 Dec 2019 17:52:03 +0100
Subject: [PATCH] luajit: switch to moonjit and update to 2.1.2

Moonjit seems to be more active in development and maintenance, which is based on LuaJIT-2.1.0-beta3. It adds ppc64le, s390x, and multiple bug fixes.

Pull request: https://github.com/openwrt/packages/pull/10718

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 lang/luajit/Makefile                   | 14 +++++++-------
 lang/luajit/patches/010-lua-path.patch |  4 ++--
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/lang/luajit/Makefile b/lang/luajit/Makefile
index 296a3ebb61d..caf51ab7608 100644
--- a/lang/luajit/Makefile
+++ b/lang/luajit/Makefile
@@ -1,13 +1,13 @@
 include $(TOPDIR)/rules.mk

 PKG_NAME:=luajit
-PKG_VERSION:=2.1.0-beta3
-PKG_RELEASE:=2
+PKG_VERSION:=2.1.2
+PKG_RELEASE:=1

-PKG_SOURCE:=LuaJIT-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://luajit.org/download
-PKG_HASH:=1ad2e34b111c802f9d0cdf019e986909123237a28c746b21295b63c9e785d9c3
-PKG_BUILD_DIR:=$(BUILD_DIR)/LuaJIT-$(PKG_VERSION)
+PKG_SOURCE:=moonjit-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/moonjit/moonjit/tar.gz/$(PKG_VERSION)?
+PKG_HASH:=c3de8e29aa617fc594c043f57636ab9ad71af2b4a3a513932b05f5cdaa4320b2
+PKG_BUILD_DIR:=$(BUILD_DIR)/moonjit-$(PKG_VERSION)

 PKG_MAINTAINER:=Morteza Milani <milani@pichak.co>
 PKG_LICENSE:=MIT
@@ -23,7 +23,7 @@ define Package/luajit
  SECTION:=lang
  CATEGORY:=Languages
  TITLE:=LuaJIT
- URL:=https://www.luajit.org
+ URL:=https://github.com/moonjit/moonjit
  DEPENDS:=@(i386||x86_64||arm||armeb||aarch64||powerpc||mips||mipsel)
 endef

diff --git a/lang/luajit/patches/010-lua-path.patch b/lang/luajit/patches/010-lua-path.patch
index f752002d22a..6abe27600c3 100644
--- a/lang/luajit/patches/010-lua-path.patch
+++ b/lang/luajit/patches/010-lua-path.patch
@@ -8,6 +8,6 @@
 -#define LUA_LUADIR	"/lua/5.1/"
 +#define LUA_LROOT	"/usr"
 +#define LUA_LUADIR	"/lua/"
- #define LUA_LJDIR	"/luajit-2.1.0-beta3/"
-
+ #define LUA_LJDIR	"/luajit-2.1.2/"
+
  #ifdef LUA_ROOT
